### ISSUE TITLE
response to multi-variable get has duplicated values

### ISSUE THREAD

**Issue by wfaulk** [lextudio/pysnmp#152]
### Expected behavior

I issued a `get_cmd()` with multiple variables in the same MIB table. I expect to get back multiple responses, each with the appropriate value.

### Actual behavior

If a `get_cmd()` is issued with multiple requested variables in the same MIB table, the value of one of the elements of the table is duplicated to all of the other elements of the table.

### Detailed steps

Testing against the simulation service, here's the result of a command-line `snmpget` command:

```
% snmpget -v 2c -c public demo.pysnmp.com IF-MIB::ifInOctets.1 IF-MIB::ifInOctets.2 IF-MIB::ifSpeed.1 IF-MIB::ifSpeed.2 IF-MIB::ifOutOctets.1 IF-MIB::ifOutOctets.2 IF-MIB::ifOutUcastPkts.1 IF-MIB::ifOutUcastPkts.2 IF-MIB::ifDescr.1 IF-MIB::ifDescr.2
IF-MIB::ifInOctets.1 = Counter32: 25611194
IF-MIB::ifInOctets.2 = Counter32: 2570427550
IF-MIB::ifSpeed.1 = Gauge32: 4294967295
IF-MIB::ifSpeed.2 = Gauge32: 2755359744
IF-MIB::ifOutOctets.1 = Counter32: 25611194
IF-MIB::ifOutOctets.2 = Counter32: 3256639591
IF-MIB::ifOutUcastPkts.1 = Counter32: 135187
IF-MIB::ifOutUcastPkts.2 = Counter32: 14916803
IF-MIB::ifDescr.1 = STRING: lo
IF-MIB::ifDescr.2 = STRING: eth0
```

Using the pysnmp script included below:

```
% ./demo.py
IF-MIB::ifInOctets.1 = 2570504176
IF-MIB::ifInOctets.2 = 2570504176
IF-MIB::ifSpeed.1 = 2755359744
IF-MIB::ifSpeed.2 = 2755359744
IF-MIB::ifOutOctets.1 = 3256802074
IF-MIB::ifOutOctets.2 = 3256802074
IF-MIB::ifOutUcastPkts.1 = 14917406
IF-MIB::ifOutUcastPkts.2 = 14917406
IF-MIB::ifDescr.1 = eth0
IF-MIB::ifDescr.2 = eth0
```

A network packet capture shows that the request and response over the network is correct.

Enabling debug logging in the script didn't seem to reveal anything useful to me.

### Python package information

7.1.13

### Operating system information

macOS 13.7.1

### Python information

Python 3.13.0

### (Optional) Contents of your test script

```python
#!/usr/bin/env python3

import asyncio

from pysnmp.hlapi.v3arch.asyncio import *

async def main():
    se = SnmpEngine()
    comm = CommunityData('public')
    target = await UdpTransportTarget.create(('demo.pysnmp.com', 161))
    context = ContextData()

    netiftable = [ ObjectIdentity('IF-MIB', 'ifInOctets', 1),
                   ObjectIdentity('IF-MIB', 'ifInOctets', 2),
                   ObjectIdentity('IF-MIB', 'ifSpeed', 1),
                   ObjectIdentity('IF-MIB', 'ifSpeed', 2),
                   ObjectIdentity('IF-MIB', 'ifOutOctets', 1),
                   ObjectIdentity('IF-MIB', 'ifOutOctets', 2),
                   ObjectIdentity('IF-MIB', 'ifOutUcastPkts', 1),
                   ObjectIdentity('IF-MIB', 'ifOutUcastPkts', 2),
                   ObjectIdentity('IF-MIB', 'ifDescr', 1),
                   ObjectIdentity('IF-MIB', 'ifDescr', 2) ]
    _, _, _, objects = await get_cmd(se, comm, target, context,
               *[ObjectType(x) for x in netiftable])
    for item in objects:
        print(item)

asyncio.run(main())
```

**Comment by lextudio-support** [comment]
New issues are marked as low priority by default. [Becoming our commercial customers](https://www.pysnmp.com/support#commercial-support), and then your reports are handled with higher priority after triage.

**Comment by lextudio-support** [comment]
New issues are marked as low priority by default. [Becoming our commercial customers](https://www.pysnmp.com/support#commercial-support), and then your reports are handled with higher priority after triage.

**Comment by wfaulk** [comment]
I've also discovered that this same problem exists even if you query the variables in different calls, unless the calls use distinct `SnmpEngine()` values.

And if you query the same variable multiple times with the same `SnmpEngine()`, any record you tried to keep of the old value is overwritten with the new one.

**Comment by ojnas** [comment]
Walk operations (walk_cmd and bulk_walk_cmd) are broken in the same way. Same value for all rows of a table in the response.

**Comment by wfaulk** [comment]
My sample script works with v7.1.9 and doesn't work with v7.1.10.

**Comment by MichalMoravik** [comment]
Thank you so much @wfaulk ! You saved me with that 7.1.9 suggestion!! 

@lextudio-support can you have a look at this one please?

**Comment by wfaulk** [comment]
How many people have arbitrarily upgraded this module and are unknowingly collecting inaccurate data?

**Comment by lextm** [comment]
Fixed in release 7.1.14.
